analyzer/codechecker_analyzer/analyzers/config_handler.py

```diff
@@ -132,14 +149,44 @@ def initialize_checkers(self,
                             available_profiles,
                             package_root,
                             checkers,
-                            checker_config=None,
+                            builtin_checker_profile_cfg=None,
                             cmdline_checkers=None,
                             enable_all=False):
+        """Initializes which checkers should be enabled during the analysis.
+
+        Enables checkers based on the configuration from the config files
+        checker profiles or user config from the command line.
+        Calls sys.exit(1) in case of the 'list' profile name was set.
+
+        Args:
+            available_profiles (Dict[str, str]): profile name and description
+            package_root (str): root path to the CodeChecker package
+            checkers (Sequence[Tuple[str, str]): (checker, description) list
+            builtin_checker_profile_cfg (Sequence[Dict]):
+                list of the available checker configs for an analyzer
+                (for every analyzer version)
+            cmdline_checkers (Sequence[Tuple[str, bool]): list of checker name
+                and if the checker was enabled in the command line
+            enable_all (bool): all the checker should be enabled
+
+        Returns:
+             bool: True for success, False if there was some problem
+                   during initinalization.
         """
-        Initializes the checker list for the specified config handler based on
-        given checker profiles, commandline arguments and the
-        analyzer-retrieved checker list.
-        """
+        problem_during_initialization = False
+
+        checker_config = None
+        # Filter out the profiles which should be valid for the given version
+        if builtin_checker_profile_cfg:
+            checker_config = \
+                self.__get_matching_checker_config(builtin_checker_profile_cfg)
+
+            if checker_config is None:
+                LOG.warning("No matching checker profile was found "
+                            "for the configured analyzer version.\n"
+                            "Please check the profile configuration in "
+                            "config.json.")
```

Comment:
In which `config.json` file? Where is this file is? It's hard to find for the user.